Lib/test/test_socket.py

```diff
@@ -5970,6 +5970,21 @@ def test_sendmsg_afalg_args(self):
             with self.assertRaises(TypeError):
                 sock.sendmsg_afalg(op=socket.ALG_OP_ENCRYPT, assoclen=-1)
 
+    def test_length_restriction(self):
+        # bpo-35050, off-by-one error in length check
+        sock = socket.socket(socket.AF_ALG, socket.SOCK_SEQPACKET, 0)
```

Comment:
You should close the socket somehow, no? Maybe: self.addCleanup(sock.close) or "with sock:".